src/ros_filter.cpp

```diff
@@ -2531,7 +2531,7 @@ namespace RobotLocalization
     {
       // Otherwise, we should use our target frame
       finalTargetFrame = targetFrame;
-      poseTmp.frame_id_ = (differential ? finalTargetFrame : msg->header.frame_id);
+      poseTmp.frame_id_ = msg->header.frame_id;
```

Comment:
```suggestion
      poseTmp.frame_id_ = (differential && !imuData ? finalTargetFrame : msg->header.frame_id);
```

Reply:
@RonaldEnsing sorry for the ping, especially as I took forever to get to this PR, but would you mind updating the PR with the code above so I can merge? If you don't have the cycles, let me know, and I'll take care of it. Thanks!

Reply:
Sure. I committed your suggestion to the branch of this PR.